bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java

```diff
@@ -79,16 +79,35 @@
   // HTTP transport used for created credentials to perform token-refresh handshakes with remote
   // credential servers. Initialized lazily to move the possibility of throwing
   // GeneralSecurityException to the time a caller actually tries to get a credential.
-  private static HttpTransportFactory httpTransportFactory = new HttpTransportFactory() {
-    @Override
-    public HttpTransport create() {
-      try {
-        return GoogleNetHttpTransport.newTrustedTransport();
-      } catch (Exception e) {
-        throw new RuntimeException("Could not create a GoogleNetHttpTransport: " + e.getMessage(), e);
+  private static HttpTransportFactory httpTransportFactory;
```

Comment:
This needs to be volatile as well:

http://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html